### PR TITLE
Migration to explore server

### PR DESCRIPTION
**Pull request by etiennehomer**

### CODE REVIEW

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 1]
```diff
@@ -319,4 +319,13 @@ private boolean isPrivateForNotification(UUID parentDirectoryUuid, boolean isCur
     public Flux<ElementAttributes> getElementsAttribute(List<UUID> ids) {
         return Flux.fromStream(() -> directoryElementRepository.findAllById(ids).stream().map(e -> toElementAttributes(e, 0)));
     }
+
+    public Mono<Void> sendUpdateTypeNotification(UUID elementUuid, String userId) {
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:323]: Maybe rename this by `sendUpdateDirectoryNotification`.
Which is consistent with related code. We can't know why it's about type here.
Then it's ok to call from explore-server a directory API which ask to emit a UPDATE_DIRECTORY notification.

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java [hunk 2]
```diff
@@ -117,4 +117,14 @@ public DirectoryController(DirectoryService service) {
     public ResponseEntity<Flux<ElementAttributes>> getElementsAttributes(@RequestParam("id") List<UUID> ids) {
         return ResponseEntity.ok().body(service.getElementsAttribute(ids));
     }
+
+    @PutMapping(value = "/directories/{elementUuid}/updateTypeNotification")
+    @Operation(summary = "send update type notification")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "The notification has been sent"),
+    })
+    public ResponseEntity<Mono<Void>> sendUpdateTypeNotification(@PathVariable("elementUuid") UUID elementUuid,
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/directory/server/DirectoryController.java:126]: We should rename this I think, see next comment.

**Review by sBouzols — CHANGES_REQUESTED** [review]
I think we should rename the sendUpdateTypeNotification method

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java (hunk 1 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/directory/server/DirectoryService.java]: I'm ok with that. Change done.

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java (hunk 2 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/directory/server/DirectoryController.java]: done

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java [hunk 3]
```diff
@@ -118,13 +118,13 @@ public DirectoryController(DirectoryService service) {
         return ResponseEntity.ok().body(service.getElementsAttribute(ids));
     }
 
-    @PutMapping(value = "/directories/{elementUuid}/updateTypeNotification")
+    @PutMapping(value = "/directories/{elementUuid}/notify-parent")
     @Operation(summary = "send update type notification")
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/directory/server/DirectoryController.java:122]: edit summary here ;)

**Review by sBouzols — CHANGES_REQUESTED** [review]
summary edit missing

File: src/main/java/org/gridsuite/directory/server/DirectoryController.java (hunk 3 above)
Comment (etiennehomer) [reply src/main/java/org/gridsuite/directory/server/DirectoryController.java]: Done

**Review by sBouzols — APPROVED** [review]
Fine Code review OK
No tests

**Review by geofjamg — APPROVED** [review]

**Pull request merged**